csharp/src/Apache.Arrow/C/CArrowSchemaImporter.cs

```diff
@@ -0,0 +1,285 @@
+                        {
+                            throw new InvalidDataException("Expected struct type child to be non-null.");
+                        }
+                        child_schemas[i] = new ImportedArrowSchema(_data->GetChild(i), isRoot: false);
+                    }
+
+
+                    List<Field> childFields = child_schemas.Select(schema => schema.GetAsField()).ToList();
+
+                    return new StructType(childFields);
+                }
+                // TODO: Map type and large list type
+
+                // Decimals
+                if (format.StartsWith("d:"))
+                {
+                    bool is256 = format.EndsWith(",256");
+                    string parameters_part = format.Remove(0, 2);
+                    if (is256) parameters_part.Substring(0, parameters_part.Length - 5);
+                    string[] parameters = parameters_part.Split(',');
+                    int precision = Int32.Parse(parameters[0]);
+                    int scale = Int32.Parse(parameters[1]);
+                    if (is256)
+                    {
+                        return new Decimal256Type(precision, scale);
+                    }
+                    else
+                    {
+                        return new Decimal128Type(precision, scale);
+                    }
+                }
+
+                // Timestamps
+                if (format.StartsWith("ts"))
+                {
+                    TimeUnit timeUnit = format[2] switch
+                    {
+                        's' => TimeUnit.Second,
+                        'm' => TimeUnit.Millisecond,
+                        'u' => TimeUnit.Microsecond,
+                        'n' => TimeUnit.Nanosecond,
+                        _ => throw new InvalidDataException($"Unsupported time unit for import: {format[2]}"),
+                    };
+
+                    string timezone = format.Split(':')[1];
+                    return new TimestampType(timeUnit, timezone);
+                }
+
+                // Fixed-width binary
+                if (format.StartsWith("w:"))
+                {
+                    int width = Int32.Parse(format.Substring(2));
+                    return new FixedSizeBinaryType(width);
+                }
+
+                return format switch
+                {
+                    // Primitives
+                    "n" => new NullType(),
+                    "b" => new BooleanType(),
+                    "c" => new Int8Type(),
+                    "C" => new UInt8Type(),
+                    "s" => new Int16Type(),
+                    "S" => new UInt16Type(),
+                    "i" => new Int32Type(),
+                    "I" => new UInt32Type(),
+                    "l" => new Int64Type(),
+                    "L" => new UInt64Type(),
+                    "e" => new HalfFloatType(),
+                    "f" => new FloatType(),
+                    "g" => new DoubleType(),
```

Comment:
These should all use the `.Default` property, where applicable. No reason to allocate extra objects here.